tests/trees-watered.test.ts

```diff
@@ -0,0 +1,195 @@
+		const { data, error } = await supabaseAnonClient.auth.signInWithPassword({
+			email: "user1@test.com",
+			password: "password1",
+		});
+		expect(data).toBeDefined();
+		expect(error).toBeNull();
+
+		const { data: waterings } = await supabaseAnonClient
+			.rpc("waterings_for_user", { u_id: users.userId1 })
+			.select("*");
+
+		expect(waterings).toBeDefined();
+		expect(waterings!.length).toBe(1);
+		expect(waterings![0].amount).toBe(10);
+		expect(waterings![0].username).toBe("user1");
+	});
+
+	it("should be able to select waterings of a specific tree via RPC", async () => {
+		const { data, error } = await supabaseAnonClient.auth.signInWithPassword({
+			email: "user1@test.com",
+			password: "password1",
+		});
+		expect(data).toBeDefined();
+		expect(error).toBeNull();
+
+		const { data: waterings } = await supabaseAnonClient
+			.rpc("waterings_for_tree", { t_id: "_0epuygrgg" })
+			.select("*");
+
+		expect(waterings).toBeDefined();
+		expect(waterings!.length).toBe(2);
+	});
+
+	it("should be able to delete own waterings", async () => {
+		const { data, error } = await supabaseAnonClient.auth.signInWithPassword({
+			email: "user1@test.com",
+			password: "password1",
+		});
+		expect(data).toBeDefined();
+		expect(error).toBeNull();
+
+		const { error: wateringsError } = await supabaseAnonClient
+			.from("trees_watered")
+			.delete()
+			.eq("id", userWateringId1!);
+
+		expect(wateringsError).toBeNull();
+
+		const { data: newWaterings } = await supabaseServiceRoleClient
+			.from("trees_watered")
+			.select("*");
+		expect(newWaterings).toBeDefined();
+		expect(newWaterings!.length).toBe(1);
+	});
+
+	it("should change username in waterings if username changes", async () => {
+		const { data, error } = await supabaseAnonClient.auth.signInWithPassword({
+			email: "user2@test.com",
+			password: "password2",
+		});
+		expect(data).toBeDefined();
+		expect(error).toBeNull();
+
+		const { error: profileUpdateError } = await supabaseAnonClient
+			.from("profiles")
+			.update({ username: "user2-update" })
+			.eq("id", users.userId2);
+
+		expect(profileUpdateError).toBeNull();
+
+		const { data: newWaterings } = await supabaseServiceRoleClient
+			.from("trees_watered")
+			.select("*");
+		expect(newWaterings).toBeDefined();
+		expect(newWaterings!.length).toBe(1);
+		expect(newWaterings![0].username).toBe("user2-update");
+	});
+
+	it("should set username and uuid in waterings to NULL if user gets deleted", async () => {
+		const { error: deleteError } =
+			await supabaseServiceRoleClient.auth.admin.deleteUser(users.userId2);
+		expect(deleteError).toBeNull();
+
+		const { data: newWaterings } = await supabaseServiceRoleClient
+			.from("trees_watered")
+			.select("*");
+		expect(newWaterings).toBeDefined();
+		expect(newWaterings!.length).toBe(1);
+		expect(newWaterings![0].username).toBeNull();
+		expect(newWaterings![0].uuid).toBeNull();
+	});
+});
```

Comment:
same questions regarding `signInWithPassword` in `beforeEach`, user deleting not their own watering, usage of `supabaseServiceRoleClient` instead of `supabaseAnonClient` than in `trees-adopted.test.ts`

Reply:
fixed usage of `supabaseServiceRoleClient`

at some points i still use `supabaseServiceRoleClient` to count the total number of waterings because i need the number for the tests. there, using anon client would not work because it is not (and should not be) possible to fetch all waterings via SELECT as an anon user.